### PR TITLE
aerosol: simple background via SPA-style activation, MACv2-SP-driven (#374)

### LINKED ISSUE #374: Add simple background aerosol
To support the two-moment microphysics (#341) we will need some background aerosol on which to add the (anthropogenic) MacSP plumes. The simplest thing to do first would be to implement HAM-lite from this repo: https://zenodo.org/records/14335069

### PR DESCRIPTION
**Pull request by duncanwp**
## Summary

Closes #374. The two-moment microphysics scheme tracks cloud droplet number `Nc` as a prognostic variable: every step has *sinks* (autoconversion, accretion, freezing, …) but no resolved *source*. Until now the source was a placeholder `base_cdnc * cdnc_factor` formula that doesn't actually scale with the MACv2-SP plume amplitudes the way an honest activation should. This PR swaps it for the **SPA approach** from [Lin et al. 2025](https://acp.copernicus.org/articles/25/15105/2025/), driven by the MACv2-SP plumes already on `AerosolData`.

The pipeline:

1. **AOD → CCN** via the existing AEROCOM-P1 Twomey fit in `macv2_sp.py`'s `get_CDNC`, stored on `AerosolData.Nccn` (cm⁻³, per column).
2. **CCN → activated droplet floor** via the SPA sublinear power-law (`jcm.physics.aerosol.spa`):
   `Nc_min[cm⁻³] = 2000 · (Nccn · Cf)^0.55`
3. **Floor applied each step** in the 2M scheme: `Nc ← max(Nc, Nc_min)`.

The 0.55 exponent lands inside the 0.3–0.8 observational band for `d ln Nc / d ln Nccn` — a *linear* mapping overestimates the indirect aerosol effect by ~50 %, which the sublinear fit corrects.

The 1M scheme and radiation continue to use `cdnc_factor` for the Twomey effect on cloud effective radius — those code paths are unchanged. The two aerosol summaries (`cdnc_factor` for radiation, `Nccn` for 2M activation) coexist by design.

## What's in this PR

- `jcm/physics/aerosol/spa.py` — new `spa_activated_cdnc` helper with `SPA_PREFACTOR=2000` and `SPA_EXPONENT=0.55` exposed for sensitivity studies.
- `Nccn` field on `AerosolData` (cm⁻³ per column, populated by `get_simple_aerosol`).
- `_apply_microphysics_2m_minimal` (`icon_physics.py`) now calls `spa_activated_cdnc(Nccn, cloud_fraction)` instead of `base_cdnc * cdnc_factor`.
- New **"Cloud–Aerosol Coupling (SPA activation)"** subsection in `docs/source/icon_physics.rst` — the user explicitly asked for an explainer because the rationale is non-obvious. Walks through *why we can't do proper Köhler activation*, *the AOD → CCN → Nc_min chain*, *why sublinear*, and *what the trade-offs are*.
- `spa_test.py` — covers units, zero-CCN/zero-cloud edge cases, sublinearity (4× CCN gives 2.18× droplets), the observational-slope band, broadcasting from column to per-level, and JIT-compat.

## Test plan

- [x] `JAX_PLATFORMS=cpu pytest -n 12 --ignore=jcm/physics/radiation/rrtmgp_test.py -m "not slow"` — 777 passed, 13 skipped.
- [x] `ruff check .` clean.
- [ ] Quick aquaplanet 2M run with MACv2-SP year_weight ramp (1900 vs 2000) showing Nc — and hence cloud effective radius — increases over the historical period. Locally I verified the SPA helper alone responds to the right inputs; flagging in case someone wants to spot-check the full pipeline before merge.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: jcm/physics/aerosol/spa.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+with SPA_PREFACTOR = 2000 and SPA_EXPONENT = 0.55. The slope falls in the
+0.3 ≤ d ln Nc / d ln Nccn ≤ 0.8 range constrained by observations.
+"""
+
+import jax.numpy as jnp
+
+
+# Lin et al. (2025) sublinear fit to E3SMv3 climatology.
+SPA_PREFACTOR: float = 2000.0
```
Comment (duncanwp) [inline jcm/physics/aerosol/spa.py]: Please make these parameters differentiable by including them in AerosolParameters

**Comment by duncanwp** [comment]
Pushed 832a013 addressing the review comment.

`spa_prefactor` (default 2000) and `spa_exponent` (default 0.55) are now fields on `AerosolParameters`, so they ride through the `tree_math.struct` pytree like every other tunable in `Parameters` and `jax.grad` flows through them. `spa_activated_cdnc` takes both as required arguments now; the call site in `_apply_microphysics_2m_minimal` pulls them from `parameters.aerosol.spa_prefactor / .spa_exponent`. Module-level constants are gone.

New test `test_differentiable_through_prefactor_and_exponent` pins that `jax.grad` produces finite, non-zero gradients w.r.t. both. `test_default_params_are_lin2025` pins the default values to the published fit and checks the exponent sits inside the 0.3–0.8 observational band.

253 aerosol/ICON/cloud/radiation tests pass; lint clean.

File: jcm/physics/icon/icon_physics.py [hunk 2]
```diff
@@ -806,10 +806,21 @@ def apply_microphysics_2m(
...
-    cdnc_factor = physics_data.aerosol.cdnc_factor
-    activated_cdnc = jnp.ones_like(state.temperature) * base_cdnc * cdnc_factor[jnp.newaxis, :]
+    # Aerosol-activated CDNC floor from the MACv2-SP plume CCN
+    # concentration via the SPA sublinear power-law (Lin et al. 2025;
+    # #374). Output is per-level `(nlev, ncols)` in m^-3 — the column-
+    # mean Nccn is broadcast to every level (vertical aerosol structure
+    # is not resolved by the simple-plumes scheme). The fit's prefactor
+    # and exponent come from `parameters.aerosol` so they remain
+    # differentiable for calibration work.
+    from jcm.physics.aerosol.spa import spa_activated_cdnc
```
Comment (duncanwp) [inline jcm/physics/icon/icon_physics.py]: I worry about having imports inside functional JAX code? Either way, the style of the rest of the code is to have the imports at the top of the module

**Comment by duncanwp** [comment]
Fixed in 83bc154 — hoisted `from jcm.physics.aerosol.spa import spa_activated_cdnc` to the top of `icon_physics.py` alongside the other module imports. Tests + lint clean.
